LYWSD03MMC.py

```diff
@@ -101,19 +102,29 @@ def watchDog_Thread():
 
 def thread_SendingData():
 	global previousMeasurements
+	global previousCallbacks
 	global measurements
 	path = os.path.dirname(os.path.abspath(__file__))
 
-
 	while True:
 		try:
 			mea = measurements.popleft()
+			invokeCallback = True
+
+			if mea.sensorname in previousCallbacks:
+				if args.callback_interval > 0 and (int(time.time()) - previousCallbacks[mea.sensorname] < args.callback_interval):
+					print("Callback for " + mea.sensorname + " would be within interval; don't invoke callback\n")
+					invokeCallback = False
+
+
 			if mea.sensorname in previousMeasurements:
 				prev = previousMeasurements[mea.sensorname]
 				if (mea == prev and identicalCounters[mea.sensorname] < args.skipidentical): #only send data when it has changed or X identical data has been skipped, ~10 packets per minute, 50 packets --> writing at least every 5 minutes
 					print("Measurements for " + mea.sensorname + " are identical; don't send data\n")
 					identicalCounters[mea.sensorname]+=1
-					continue
+					invokeCallback = False
+
+			if invokeCallback == False: continue
 
 			if args.callback:
```

Comment:
Just had a quick glance, but it should be better to check here 
if args.callback and invokeCallback == True:
instead of if invokeCallback == False: continue

Reply:
True. Just didn't want to change your original code too much.

Reply:
I can see the danger of thinking you're inside the loop during further developments, but then this run is interrupted via Continue. Therefore, it is better to adjust the condition for all callbacks instead of terminating the loop pass via Continue.

Reply:
True. Changed this and committed.